File: angular2-tour-of-heroes/app/app.component.ts
import {Component} from 'angular2/core';
import {HeroService} from './hero.service';
import {HeroesComponent} from './heroes.component';
import { RouteConfig, ROUTER_DIRECTIVES, ROUTER_PROVIDERS } from 'angular2/router';
import {DashboardComponent} from './dashboard.component';
import {HeroDetailComponent} from './hero-detail.component';

@Component({
  selector:'my-app',
    template: `
    <h1>{{title}}</h1>
    <nav>
    <a [routerLink]="['Heroes']">Heroes</a>
    <a [routerLink]="['Dashboard']">Dashboard</a>
    </nav>
    <router-outlet></router-outlet>`,
    directives: [ROUTER_DIRECTIVES],
    styleUrls: ['app/app.component.css'],
    providers:[ROUTER_PROVIDERS,HeroService]
})
@RouteConfig([{
  path: '/heroes',
  name: 'Heroes',
  component: HeroesComponent
},
  {
    path: '/dashboard',
    name: 'Dashboard',
    component: DashboardComponent,
    useAsDefault: true
  },
{
  path: 'detail/:id',
  name: 'HeroDetail',
  component: HeroDetailComponent
}])
export class AppComponent {
  title = 'Tour of Heroes';
}

File: dtour/lib/app_component.dart
// Copyright (c) 2016, <your name>. All rights reserved. Use of this source code
// is governed by a BSD-style license that can be found in the LICENSE file.

import 'package:angular2/core.dart';
import 'package:dtour/hero-detail_component.dart' show HeroDetailComponent;
import 'package:dtour/Hero.dart' show Hero;
import 'package:dtour/hero_service.dart';

@Component(
    selector: 'my-app',
    templateUrl: 'app_component.html',
    styleUrls: const ['app_component.css'],
    directives: const [HeroDetailComponent],
    providers: const [HeroService])
class AppComponent implements OnInit {
  var title = 'Tour of Heroes';
  List<Hero> heroes;
  Hero selectedHero;
  final HeroService _heroService;
  AppComponent(this._heroService);

  getHeroes() async {
    heroes = await _heroService.getHeroes();
  }

  void onSelect(Hero hero) {
    selectedHero = hero;
  }

  ngOnInit() {
    getHeroes();
  }
}
